frontend/src/components/widgets/DataFrame/DataFrame.tsx

```diff
@@ -220,7 +220,7 @@ function DataFrame({
         setNumRows(editingState.current.getNumRows())
       }
     }
-  }, [])
+  }, [widgetMgr, originalColumns, element, READ_ONLY])
```

Comment:
I'm a bit worried that this has any strange side effect since this Is supposed to be only run at the first rendering. But I assume all these dependencies should stay stable anyways

Reply:
Hey good point Lukas. Do you have any recommendations as to what sanity testing I should maybe do in order to check this? I was thinking that e2e tests and unit tests would be good enough but if there's any other manual testing, I would love to test that out.

Reply:
I dont have too many concrete things, just a bit of a bad feeling with this one line. Maybe it could behave a bit strangely with `add_rows`... but I think we are probably fine here.

Reply:
We could either merge it and hoping it goes well. Or we could also ignore this line for know and I have another look when I'm back. I'm fine with either way

Reply:
This seems like a potentially good solution https://stackoverflow.com/a/56767883/8576456 - it may require us to suppress the particular eslint warning at the declaration site, but it makes the intention very clear

Reply:
@willhuang1997 I think if Lukas is worried about potential unexpected side effects, we should roll back this change before merging (either with a `useMountEffect` helper like the one mentioned in that stackoverflow,  or with an explicit eslint ignore line with a comment about why we're ok with the warning suppression)

Reply:
Added a comment for now and will let Lukas or me look at this later!